tests/test_clnrest.py

```diff
@@ -488,3 +499,31 @@ def test_clnrest_old_params(node_factory):
     assert [p for p in l2.rpc.plugin('list')['plugins'] if p['name'].endswith('clnrest')] == []
     assert l2.daemon.is_in_log(r'plugin-clnrest: Killing plugin: disabled itself at init: `clnrest-port` option is not configured')
     assert l2.daemon.is_in_log(rf'clnrest-use-options.py: rest-port is {rest_port}')
+
+
+def test_clnrest_websocket_upgrade_header(node_factory):
+    """Test that not setting an upgrade header leads to rejection"""
```

Comment:
The comment may be misleading i think. It is possible to call the http endpoints without using a websocket. In fact the websocket in clnrest is only for listening to notifications. 

I wrote this test because in order to differentiate what a client wants when calling the root path i'm checking the `upgrade` header. Every websocket client i tested (RTL + ZEUS) included that but not the client used in the tests.

Not setting this header (as in this test) makes the request look like a regular http request and in this case will return a redirect to the swagger ui which the socketio client interprets as an unexpected response.

I'd say in practice this does not make a change in behaviour since a normal websocket client sets the `upgrade` header.

Reply:
I only wrote this tests because i had a bug in an earlier version where not setting the upgrade header would circumvent the rune check for notifications.